vetiver/tests/test_add_endpoint.py

```diff
@@ -13,31 +21,50 @@ def _start_application(check_ptype):
         description="A regression model for testing purposes",
     )
 
-    def sum_values(x):
-        return x.sum()
+    return v
+
 
-    app = VetiverAPI(v, check_ptype=check_ptype)
+def sum_values(x):
+    return x.sum()
 
+
+@pytest.fixture
+def vetiver_client(vetiver_model):  # With check_ptype=True
+
+    app = VetiverAPI(vetiver_model, check_ptype=True)
     app.vetiver_post(sum_values, "sum")
 
-    return app
+    app.app.root_path = "/sum"
+    client = TestClient(app.app)
 
+    return client
 
-def test_endpoint_adds_ptype():
-    app = _start_application(check_ptype=True).app
 
-    client = TestClient(app)
-    data = {"B": 0, "C": 0, "D": 0}
-    response = client.post("/sum", json=data)
-    assert response.status_code == 200, response.text
-    assert response.json() == {"sum": [0]}, response.json()
+@pytest.fixture
+def vetiver_client_check_ptype_false(vetiver_model):  # With check_ptype=False
+
+    app = VetiverAPI(vetiver_model, check_ptype=False)
+    app.vetiver_post(sum_values, "sum")
+
+    app.app.root_path = "/sum"
+    client = TestClient(app.app)
+
+    return client
+
+
+def test_endpoint_adds_ptype(vetiver_client):
+
+    data = pd.DataFrame({"B": [1, 1, 1], "C": [2, 2, 2], "D": [3, 3, 3]})
+    response = vetiver.predict(endpoint=vetiver_client, data=data)
+
+    assert isinstance(response, pd.DataFrame)
+    assert response.to_json() == '{"sum":{"0":3,"1":6,"2":9}}', response.to_json()
+
 
+def test_endpoint_adds_no_ptype(vetiver_client_check_ptype_false):
 
-def test_endpoint_adds_no_ptype():
-    app = _start_application(check_ptype=False).app
+    data = pd.DataFrame({"B": [1, 1, 1], "C": [2, 2, 2], "D": [3, 3, 3]})
+    response = vetiver.predict(endpoint=vetiver_client_check_ptype_false, data=data)
```

Comment:
Because data is sent as a `Request` with no ptype, this will not be reconstructed as a DataFrame (an endpoint with a ptype registered will do this). Does this make sense, or should there be a stronger handoff to a dataframe?

Reply:
After thinking about this, I believe this is the expected behavior (essentially taking in whatever is given, as-is).